include/rocksdb/options.h

```diff
@@ -1514,6 +1514,21 @@ struct SizeApproximationOptions {
   // Defines whether the returned size should include data serialized to disk.
   // If set to false, include_memtabtles must be true.
   bool include_files = true;
+  // When approximating the files total size that is used to store a keys range
+  // (using DB::GetApproximateSizes), we first sum up the sizes of the files
+  // that fully fall into the range. Then we sum up the sizes of all the
+  // files that may intersect the range (this includes all files in L0 as well).
+  // Then, if intersecting_files_size is smaller than
+  // (full_files_size / full_to_intersecting_file_size_ratio) - we we can infer
+  // that the intersecting files have a sufficiently negligible contribution to
+  // the total size, and we can approximate the storage required for the keys in
+  // range as just half of the intersecting_files_size. E.g., if the value
+  // of full_to_intersecting_file_size_ratio is 10, then the error of the
+  // approximation is limited to only ~10% of the total size of files that fully
+  // fall into the keys range. In such case, this helps to avoid a costly
+  // process of binary searching the intersecting files that is required only
+  // for a more precise calculation of the total size.
+  uint32_t full_to_intersecting_file_size_ratio = 0;
```

Comment:
Sorry for having a new comments now. "Intersecting file" is an implementation detail, which I hope we can hide behind the option name, so that in the future we have more flexibility. Some more general name like, accuracy ratio, or something like that, would be better.
I also prefer the user facing comments describe more about the results, e.g. their result is at most x times off the most accurate estimation. The implementation detail comment is awesome but is best to locate at internal source files, like DBImpl or VersionSet.

Reply:
Hmmm....
1. What name would you suggest?
2. What comments would you like me to add? In a sense - it would be a bit hard to explain the meta-physical meaning of this option while staying vague. 
Please advise.

Reply:
How about a vague name like `approximate_ratio`?
As comments, we can go as vague as with this option, RocksDB is allowed to take approximation up to this ratio. If you want, we can go one step further, saying we don't dive into some SST file. I don't think we should say we first do what and second do what. This is implementation details and will be applicable to change.

Also, do we consider to use a double value, which the inaccuracy is calculated as `full_files_size * the_ratio`? My concern with the current int value is that, it cannot go < 1.0. If the user is OK with very approximate number, like intersecting files can be skipped if it is within 2x of the total size for full files, we aren't able to provide this information using this current option.

Reply:
I went with an int on purpose. The issue with double type is that it can be:
- negative
- positive
- > 1.0
- < 1.0

Also comparison to 0.0 should be done with care.
In other words - it gives too many degrees of freedom at the API level, without clear benefit to the precision or the functionality.
Maybe I am wrong, but it feels that there is also no clearly good value for this option. Something like 5 to 10 (giving ~10% precision makes sense). Everything else either yields useless approximate-size results or doesn't trigger the heuristic.

I can change it to double, of course, if you insist. Just wanted to lay out some reasoning.

Reply:
@elipoz what you said makes sense too. If you feel strong about it, we can keep this int value.